manifests/unit_file.pp

```diff
@@ -9,6 +9,25 @@
 ) {
   include ::systemd
 
+  if $title =~ /^([^\/]+)\// {
+    $subdir = $1
+    if $subdir !~ /\.d$/ {
+      fail('drop-in unit dirs must have suffix .d')
+    }
+    if $title !~ /\.conf$/ {
+      fail('drop-in unit files must have suffix .conf')
+    }
+    file { "${path}/${subdir}":
```

Comment:
@rj667 I'd rather avoid an exec resource. If you have to take that road, I'd rather you use `ensure_directory()`